### PR TITLE
FIx flask-cors and pymongo vulnerabilities

### PR DESCRIPTION
**Pull request by jrobinAV**
### Description

This PR is about patching taipy 3.1.1 to fix the vulnerabilities. It is about merging it in the `release/3.1` branch.

### Done

- Change version number
- Update dependencies: pymongo, flask-cors, numpy
- Remove Python 3.8 support because the new dependencies do not support it.
- Fix Ruff complaints about Type comparison using `==` instead of `is`
- Fix gui/gui_specific/test_gui.py

### CODE REVIEW

File: INSTALLATION.md [hunk 1]
```diff
@@ -5,7 +5,7 @@ There are different ways to install Taipy, depending on how you plan to use it.
 If your goal is to look into the code, modify and improve it, go straight
 to the [source installation](#installing-for-development) section.
 
-Taipy needs your system to have Python 3.8 or above installed.
+Taipy needs your system to have Python 3.3 or above installed.
```
Comment (FredLL-Avaiga) [inline INSTALLATION.md]: 3.9

Comment (jrobinAV) [reply INSTALLATION.md]: Thx

File: taipy/config/common/_template_handler.py [hunk 2]
```diff
@@ -51,11 +51,11 @@ def _replace_template(cls, template, type, required, default):
...
-            if type == bool:
+            if type is bool:
                 return cls._to_bool(val)
-            elif type == int:
+            elif type is int:
                 return cls._to_int(val)
-            elif type == float:
+            elif type is float:
                 return cls._to_float(val)
             elif type == Scope:
```
Comment (FredLL-Avaiga) [inline taipy/config/common/_template_handler.py:60]: why not `is` ?

File: taipy/gui/data/array_dict_data_accessor.py [hunk 3]
```diff
@@ -33,14 +33,14 @@ def __get_dataframe(self, value: t.Any) -> t.Union[t.List[pd.DataFrame], pd.Data
...
                     lengths = {len(x) for x in value}
                     return (
                         pd.DataFrame(value)
                         if len(lengths) == 1
                         else [pd.DataFrame({f"{i}/0": v}) for i, v in enumerate(value)]
                     )
-                elif type_elt == dict:
+                elif type_elt is dict:
                     return [pd.DataFrame(v) for v in value]
                 elif type_elt == _MapDict:
```
Comment (FredLL-Avaiga) [inline taipy/gui/data/array_dict_data_accessor.py:45]: why not is ?

File: taipy/config/common/_template_handler.py (hunk 2 above)
Comment (jrobinAV) [reply taipy/config/common/_template_handler.py:60]: These are Ruff auto changes... I don't know the precise rule.

File: taipy/gui/data/array_dict_data_accessor.py (hunk 3 above)
Comment (jrobinAV) [reply taipy/gui/data/array_dict_data_accessor.py:45]: These are Ruff auto changes... I don't know the precise rule.

Comment (FredLL-Avaiga) [reply taipy/gui/data/array_dict_data_accessor.py:45]: 😶‍🌫️🤔😮😱

Comment (FredLL-Avaiga) [reply taipy/gui/data/array_dict_data_accessor.py:45]: "The == operator compares the value or equality of two objects, whereas the Python is operator checks whether two variables point to the same object in memory."
means that a type should be compared with `is`

Comment (jrobinAV) [reply taipy/gui/data/array_dict_data_accessor.py:45]: This was the initial error fixed by ruff:
https://docs.astral.sh/ruff/rules/type-comparison/

Comment (jrobinAV) [reply taipy/gui/data/array_dict_data_accessor.py:45]: Does not help understand why it does not complain about `type_elt == _MapDict:`

**Comment by jrobinAV** [comment]
@FredLL-Avaiga ANy help on the test failure would help me a lot.

File: taipy/gui/data/array_dict_data_accessor.py (hunk 3 above)
Comment (FredLL-Avaiga) [reply taipy/gui/data/array_dict_data_accessor.py:45]: `is` is more efficient but `==` is valid too

**Review by FabienLelaquais — DISMISSED** [review]
👌

**Review by FabienLelaquais — APPROVED** [review]
👌

**Pull request merged**